### PR TITLE
cmd/cored: introduce loopback_auth build tag

### PR DESCRIPTION
**Pull request by boymanjor**
By default, cored will authenticate requests made on the loopback
interface. To allow unauthenticated requests on this interface cored
should be built with the `loopback_auth` build tag. This is a part
of the work started in #859.

### CODE REVIEW

**Comment by boymanjor** [comment]
screen shot 2017-04-10 at 2 22 43 pm

**Comment by boymanjor** [comment]
PTAL

File: cmd/cored/prod.go [hunk 1]
```diff
@@ -15,6 +14,10 @@ var prod = true
 
 func resetInDevIfRequested(db pg.DB) {}
 
-func authLoopbackInDev(req *http.Request) bool {
-	return false
+func devEnableMockHSM(_ pg.DB) []core.RunOption {
+	return nil
+}
+
+func devHSM(_ pg.DB) (blocksigner.Signer, error) {
```
Comment (kr) [inline cmd/cored/prod.go]: I think these are left over from another rebase?

Comment (boymanjor) [inline cmd/cored/prod.go]: Yep, thanks.

**Comment by boymanjor** [comment]
Updated, PTAL.

File: cmd/cored/no_loopback_authn.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+//+build no_loopback_authn
```
Comment (kr) [inline cmd/cored/no_loopback_authn.go]: Normally I am all on board with using "authn" or "authz" to distinguish. However, in this case @tessr will shortly be doing the authz project and as part of that I'm pretty sure this flag will change from modifying authn behavior to modifying authz behavior.

We could update the tag name at that time… or we could cheat and just call it "auth" now. :stare:

File: cmd/deploy/main.go [hunk 3]
```diff
@@ -87,7 +87,7 @@ func mustBuild(filename string) []byte {
 	commit = bytes.TrimSpace(commit)
 	date := time.Now().UTC().Format(time.RFC3339)
 	cmd = exec.Command("go", "build",
-		"-tags", "plain_http",
+		`-tags "plain_http no_loopback_authn"`,
```
Comment (kr) [inline cmd/deploy/main.go]: Ah, same thing about the quote marks here.

**Comment by kr** [comment]
Review'd

File: cmd/cored/no_loopback_authn.go (hunk 2 above)
Comment (boymanjor) [inline cmd/cored/no_loopback_authn.go]: I'm fine with using "auth" now.

**Comment by boymanjor** [comment]
I updated this to use "auth" instead of "authn" and fixed the tags for `benchcore` and `deploy`. PTAL

**Comment by boymanjor** [comment]
PTAL

File: bin/build-cored-release [hunk 4]
```diff
@@ -48,7 +48,7 @@ DATE=${DATE:-`date +%s`} # date can be set via envvar
 ldflags="-X main.buildTag=$releaseRef -X main.buildCommit=$commit -X main.buildDate=$DATE"
 
 go build\
-  -tags 'plain_http'\
+  -tags 'plain_http loopback_auth'\
```
Comment (kr) [inline bin/build-cored-release]: Do we want this tag here? If this is for a production server, we should leave it out right?

Comment (boymanjor) [inline bin/build-cored-release]: I thought we were only using this script for developer edition builds. If not you're correct. @jeffomatic or @dominic any insight?

Comment (boymanjor) [inline bin/build-cored-release]: Discussed out of band: the product folks are using this for developer edition builds. A search of the codebase shows it only being used for this purpose as well. Should I add a comment about the flags here?

Comment (kr) [inline bin/build-cored-release]: Nah, I think it's good as-is. We are prob going to be reworking the release tools soon anyway.

**Pull request merged**